PayloadPkg/OsLoader/LoadImage.c

```diff
@@ -344,15 +401,41 @@ GetTraditionalLinux (
     DEBUG ((DEBUG_INFO, "Checking %s\n",mConfigFileName[Index]));
     ConfigFile     = NULL;
     ConfigFileSize = 0;
-    Status = GetFileByName (FsHandle, (CHAR16 *)mConfigFileName[Index], &ConfigFile, &ConfigFileSize);
+
+    Status = OpenFile (FsHandle, (CHAR16 *)mConfigFileName[Index], &FileHandle);
+    if (EFI_ERROR (Status)) {
+      DEBUG ((DEBUG_INFO, "Open file '%s' failed, Status = %r\n", (CHAR16 *)mConfigFileName[Index], Status));
+      continue;
+    }
+
+    Status = GetFileSize (FileHandle, &ConfigFileSize);
+    if (EFI_ERROR (Status)) {
+      DEBUG ((DEBUG_INFO, "Get file '%s' size failed, Status = %r\n", (CHAR16 *)mConfigFileName[Index], Status));
+      CloseFile (FileHandle);
+      continue;
+    }
+
+    ConfigFile = AllocatePool (ConfigFileSize);
+    if (ConfigFile == NULL) {
+      CloseFile (FileHandle);
+      return EFI_OUT_OF_RESOURCES;
+    }
+
+    Status = ReadFile (FileHandle, &ConfigFile, &ConfigFileSize);

```

Comment:
we should not allocate memory every time when reading file. In the file system instance (e.g. ExtFsReadFile), the memory was allocated there.  or just allocate in caller, not allocate memory in file system instance.

Reply:
Good catch. This is ExtFsReadFile bug. Let me remove memory allocation in ExtFsReadFile.

Reply:
It is fine to fix it in another patch.